lib/gollum-lib/page.rb

```diff
@@ -466,6 +506,28 @@ def page_match(name, path)
       false
     end
 
+    # Private: get the node where the header is located
+    #
+    # Extracts the header node from doc.
+    #
+    # TODO: this case is a bad encapsulation, the markup
```

Comment:
Agreed, we don't want markup-specific code here. How much work would it be to change that?